src/flows/context/flow.current.tsx

```diff
@@ -69,37 +69,45 @@ export const FlowProvider: FC = ({children}) => {
   }, [flows, currentID])
 
   useEffect(() => {
-    if (isFlagEnabled('sharedFlowEditing')) {
+    const doc = yDoc.current
+    if (isFlagEnabled('sharedFlowEditing') && currentID) {
       provider.current = new WebsocketProvider(
         'wss://demos.yjs.dev', // todo(ariel): replace this with an actual API that we setup
-        flowId, // todo(ariel): we might need to confine this to an org, depends on how multi-org plays out
-        yDoc.current
+        currentID, // todo(ariel): we might need to confine this to an org, depends on how multi-org plays out
+        doc
       )
       provider.current.on('sync', isSynced => {
         if (isSynced) {
-          const localState = yDoc.current.getMap('localState')?.toJSON()
-          if (!localState) {
-            yDoc.current.getMap('localState').set('localState', currentFlow)
+          const {flowUpdateData} = doc.getMap('flowUpdateData')?.toJSON()
+          if (!flowUpdateData && currentFlow) {
+            doc
+              .getMap('flowUpdateData')
+              .set('flowUpdateData', serialize(currentFlow, orgID))
```

Comment:
we shouldn't be filling in the orgID here. and more generally, why do we need it?

Reply:
The serializer requires it